File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.rs

impl Solution {
    pub fn find_min(nums: Vec<i32>) -> i32
    {
        let (mut i, mut j) = (0, nums.len()-1);

        while i < j
        {
            let k = (i+j) / 2;

            if nums[k] > nums[j] {
                i = k+1;
            } else {
                j = k;
            }
        }

        nums[i]
    }
}

File: Data Structures & Algorithms/jump-game/submission-1.cpp

class Solution {
public:
    bool canJump(vector<int>& nums)
    {
        if (nums.empty()) {
            return true; // vacuous truth
        }

        auto dp = vector<bool>(nums.size(),false);
        dp[0] = true;

        for (auto i = 0; i < nums.size(); ++i)
            if (dp[i])
                for (auto j = 1; j < std::min(nums[i]+1,(int)nums.size()-i); ++j)
        {
            dp[i+j] = true;
        }

        return dp[nums.size()-1];
    }
};
